com.unity.render-pipelines.core/Editor/Utilities/MenuUtilities.cs

```diff
@@ -0,0 +1,159 @@
+            //   - each package add its own attribute (one with renderPipelineAssetTypes = { HDRPAsset }, other with renderPipelineAssetTypes = { URPAsset })
+            var filteredMethods = TypeCache.GetMethodsWithAttribute<MenuItemForRenderPipeline>();
+            foreach (var methodInfo in filteredMethods)
+            {
+                var attribute = methodInfo.GetCustomAttribute<MenuItemForRenderPipeline>();
+                foreach (Type rpAssetType in attribute.renderPipelineAssetTypes)
+                {
+                    if (!s_MenuMap.TryGetValue(rpAssetType, out var menuItemsForRP))
+                    {
+                        menuItemsForRP = new List<MenuItemData>();
+                        s_MenuMap[rpAssetType] = menuItemsForRP;
+                    }
+
+                    var query = menuItemsForRP.Where(mi => mi.attribute.path == attribute.path);
+                    bool alreadyHere = query.Count() > 0;
+                    var menuItemData = alreadyHere ? query.First() : new MenuItemData();
+
+                    menuItemData.attribute = attribute;
+                    if (attribute.validate)
+                        menuItemData.validate = (Func<bool>)Delegate.CreateDelegate(typeof(Func<bool>), methodInfo);
+                    else
+                        menuItemData.action = (Action)Delegate.CreateDelegate(typeof(Action), methodInfo);
+
+                    if (!alreadyHere)
+                        menuItemsForRP.Add(menuItemData);
+                }
+            }
+
+            //activating menuitems
+            UpdateMenuItemForRenderPipeline();
+        }
+
+        static void UpdateMenuItemForRenderPipeline()
+        {
+            // remove
+            foreach (var menuItemData in s_MenuMap.Values.SelectMany(mid => mid))
+                UnregisterMenu(menuItemData); //do nothing if not registered before
+
+            // add if required
+
+            // Note: GraphicsSettings.currentRenderPipeline == null possible and should mean built-in
+            // though we cannot use null as key so it was early replaced by typeof(RenderPipelineAsset)
+            Type currentSRPType = GraphicsSettings.currentRenderPipeline?.GetType() ?? typeof(RenderPipelineAsset);
+            if (!s_MenuMap.TryGetValue(currentSRPType, out var menuItemDatasForRP))
+                return;
+
+            foreach (var menuItemData in menuItemDatasForRP)
+                RegisterMenu(menuItemData);
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Method)]
+    public class MenuItemForRenderPipeline : Attribute
+    {
+        public readonly Type[] renderPipelineAssetTypes;
+        public readonly string path;
+        public readonly int priority;
+        public readonly bool validate;
+
+        public MenuItemForRenderPipeline(string path, bool validate = false, int priority = 0, params Type[] renderPipelineAssetTypes)
```

Comment:
Are we sure we want the asset and not the pipeline?

Reply:
Jenny seamed to say that the only thing we are certain at editor start is GraphicsSettings.currentRenderPipeline which is the Asset. Did I miss something @jenniferd-unity ? I would also prefer checking the RenderPipeline and not the asset if possible